execution/utility.sh

```diff
@@ -3418,95 +3416,59 @@ function create_containter_registry_repository {
-  info "  - sha1: $(cat ${local_file}.sha1 )"
+  info "  - sha1: $(cat "${local_file}.sha1" )"
   info "  - local file: ${registry_file_name}"
   return 0
 }
 
 function get_url_image_to_registry() {
   local source_url="$1"; shift
   local expected_hash="$1"; shift
-  local image_format="$1"; shift
-  local registry_region="$1"; shift
-  local registry_bucket="$1"; shift
-  local registry_path="$1"; shift
-  local registry_file_name="$1"; shift
-  local product="$1"; shift
-  local environment="$1"; shift
-  local segment="$1"; shift
-  local build_unit="$1"; shift
+  local registry_location="${1}"; shift
+  local image_file_name="$1"; shift
   local zip_image_content="${1}"; shift
 
   pushTempDir "hamlet_imageUrl_XXXXXX"
-  local local_dir="$(getTopTempDir)"
+  local_dir="$(getTopTempDir)"
 
-  get_image_from_url "${source_url}" "${local_dir}" "${registry_file_name}"
+  get_image_from_url "${source_url}" "${local_dir}" "${image_file_name}"
 
   if [[ "${zip_image_content}" == "true" && "$(fileExtension "${source_url}")" != "zip" ]]; then
-    local build_reference="$(cat ${local_dir}/$(fileName "${source_url}").sha1 )"
-    ( cd "${local_dir}" && zip -j "${local_dir}/${registry_file_name}" "${local_dir}/$(fileName "${source_url}")")
-    sha1sum "${local_dir}/${registry_file_name}" | cut -d " " -f 1  > "${local_dir}/${registry_file_name}.sha1"
+    image_hash="$(cat "${local_dir}/$(fileName "${source_url}").sha1" )"
+    ( cd "${local_dir}" && zip -j "${local_dir}/${image_file_name}" "${local_dir}/$(fileName "${source_url}")")
+    sha1sum "${local_dir}/${image_file_name}" | cut -d " " -f 1  > "${local_dir}/${image_file_name}.sha1"
   else
-    local build_reference="$(cat ${local_dir}/${registry_file_name}.sha1 )"
+    image_hash="$(cat "${local_dir}/${image_file_name}.sha1" )"
   fi
 
-  if [[ -n "${expected_hash}" && -n "${build_reference}" ]]; then
-    if [[ "${build_reference}" != "${expected_hash}" ]]; then
-      fatal "Image from url: ${source_url} sha1: ${build_reference} does not match expected sha1 hash: ${expected_hash}"
+  if [[ -n "${expected_hash}" ]]; then
+    if [[ "${image_hash}" != "${expected_hash}" ]]; then
+      fatal "Image from url: ${source_url} sha1: ${image_hash} does not match expected sha1 hash: ${expected_hash}"
       return 255
     fi
   fi
 
   popTempDir
 
-  update_build_reference_from_image "${product}" "${environment}" "${segment}" "${build_unit}" "${build_reference}" "${image_format}" "${source_url}"
-
-  info "Uploading image to registry..."
-  if [[ -n "${build_reference}" && -f "${local_dir}/${registry_file_name}" ]]; then
-    aws --region "${registry_region}" s3 sync --no-progress --delete "${local_dir}" "s3://${registry_bucket}/${registry_path}/${build_reference}/"
+  info "Uploading image to registry ${registry_location}/${expected_hash}"
+  if [[ -f "${local_dir}/${image_file_name}" ]]; then
+    aws s3 sync --no-progress --delete "${local_dir}" "${registry_location}/${expected_hash}"
```

Comment:
Are we changing from using repo commit ids to using hashes? Don't quite understand what the intent is here...

Reply:
This is only when pulling the image from a URL that has been provided, not for images uploaded manually ( i.e. an image built and uploaded using upload-image )

Reply:
ok